runFetchDataAnalyseNew.py

```diff
@@ -0,0 +1,202 @@
+addAttributes = cfgMain['FILTERING']['addAttributes'].split('|')
+resampleDist =  cfgMain['FILTERING'].getfloat('resampleDist')
+
+# Start logging
+log = logUtils.initiateLogger(avalancheDir, logName, modelInfo='AmaConnector')
+log.info('MAIN SCRIPT')
+log.info('Current search: %s', avalancheDir)
+
+# fetch all info of the database according to queryString and return dataFrame
+dbData = gD.grabAllComplete(avalancheDir, queryString=queryString, accessfile=accessfile)
+log.info('Fetched %d entries from data base ' % (len(dbData)))
+for index, row in dbData.iterrows():
+    log.info('%s, event id: %s, index %s' % (row['path_name'], row['event_id'], index))
+
+
+
+# filter db according to nonEmptyCols and convert geometry entries to desired projection
+# resample thalweg for higher resolution
+dbFiltered = aU.fetchGeometryInfo(dbData, 'epsg:4326', cfgMain['MAIN']['projstr'],
+    cfgMain['FILTERING']['geomStr'], nonEmptyCols, addAttributes, resampleDist)
+
+log.info('Filtered db data and converted to %s' % cfgMain['MAIN']['projstr'])
+log.info('Events found for path name:')
+for index, row in dbFiltered.iterrows():
+    log.info('%s, event id: %s' % (row['path_name'], row['event_id']))
+
+#Import settings from .ini-File
+projstr = cfgMain['MAIN']['projstr']
+resDist = cfgMain['MAIN'].getfloat('resamplePathFit')
+dsMin = cfgMain['MAIN'].getfloat('dsMin')
+slope1 = cfgMain['MAIN'].getfloat('slopeAngle1')
+slope2 = cfgMain['MAIN'].getfloat('slopeAngle2')
+
+mask = dbFiltered.duplicated(subset='path_id', keep ='first')
+dbFiltered = dbFiltered[~mask]
+
+dbFiltered.dropna(subset=['geom_rel_event_pt3d_epsg:31287'], inplace=True)
+
+
+# snap release, runout, origin, transit, deposition points to resampled thalweg for all events found that match criteria
+dbFiltered = gT.snapPtsToLine(dbFiltered, projstr, lineName='geom_path_ln3d',
+    pointsList=[ 'geom_event_pt3d', 'geom_origin_pt3d',
+        'geom_transit_pt3d', 'geom_runout_pt3d', 'geom_event_pt3d', 'geom_rel_event_pt3d'])
+
+# compute distance along thalweg between rel- runout, orig-transit, orig-depo
+# here the snapped points are chosen also in terms of their elevation on the path line!
+
+dbFiltered = aU.addXYDistAngle(dbFiltered, 'geom_path_ln3d_%s_resampled' % projstr, 'geom_origin_pt3d_%s_snapped' % projstr,
+    'geom_transit_pt3d_%s_snapped' % projstr, projstr, name='orig-transit')
+
+dbFiltered = aU.addXYDistAngle(dbFiltered, 'geom_path_ln3d_%s_resampled' % projstr, 'geom_origin_pt3d_%s_snapped' % projstr,
+    'geom_runout_pt3d_%s_snapped' % projstr, projstr, name='orig-depo')
+
+dbFiltered = aU.addXYDistAngle(dbFiltered, 'geom_path_ln3d_%s_resampled' % projstr, 'geom_origin_pt3d_%s_snapped' % projstr,
+                               'geom_event_pt3d_%s_snapped'% projstr, projstr, name='orig-runout')
+
+dbFiltered = aU.addGradientForPoint(dbFiltered, 'geom_path_ln3d_%s_resampled' % projstr, ['geom_origin_pt3d_%s_snapped' % projstr, 
```

Comment:
add comment how this is done - so far in this function the distance over 6 points (if available +-3 before and after point of interest) (dependent on resampling distance of path line) is used to compute slope angle - hardcoded in function

Reply:
Should the "6" go to the ini?

Reply:
set as TODO